ibis/examples/__init__.py

```diff
@@ -38,7 +38,7 @@ def fetch(
         name = self.name
 
         if table_name is None:
-            table_name = ibis.util.gen_name(f"examples_{name}")
```

Comment:
Do we have any SQL backends where creating a temporary table with the same name as a non-temporary table will clobber the non-temporary table? If so, dropping the UUID suffix here could result in data loss if the user has any tables that match the example name.

Reply:
We might also decide this isn't a problem we care about (what user would load example data into a production database?). On further thought even if the above is true for some backends, IMO the risk here is low and the benefits of consistent naming feel worth it.

Reply:
AFAIK all of our backends that support temporary tables either fail with an error about already existing tables, or implement shadowing, where the temporary table takes precedence over the non-temporary when resolving table names.

Reply:
DuckDB (for example) implements shadowing:

```
D create table t as select 1 as a;
D create temp table t as select 1 as b;
D table t;
┌───────┐
│   b   │
│ int32 │
├───────┤
│     1 │
└───────┘
D drop table t;
D table t;
┌───────┐
│   a   │
│ int32 │
├───────┤
│     1 │
└───────┘
```